### PR TITLE
Improve flamegraph profiling instructions

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?
Related to https://github.com/apache/datafusion/issues/11682

## Rationale for this change

While working on https://github.com/apache/datafusion/pull/12092 I found myself needing to make flamegraphs and the instructions were a bit lackign.

I also made a short (5 min) video to help this process along https://youtu.be/2z11xtYw_xs that I wanted to link as well

## What changes are included in this PR?

Update the flamegraph creation cookbook. Here is how it looks rendered:

Screenshot 2024-09-18 at 9 31 15 AM

## Are these changes tested?
I tested them locally

## Are there any user-facing changes?
Better docs

### CODE REVIEW

**Review by hiltontj — APPROVED** [review]

**Comment by alamb** [comment]
FYI @Rachelint

**Comment by Rachelint** [comment]
Nice! I am double checking the steps in docs in my local.

File: docs/source/library-user-guide/profiling.md [hunk 1]
```diff
@@ -25,34 +25,44 @@ The section contains examples how to perform CPU profiling for Apache DataFusion
...
+in images such as this:
 
-```bash
-CARGO_PROFILE_RELEASE_DEBUG=true cargo flamegraph --root --unit-test datafusion  -- dataframe::tests::test_array_agg
+![Flamegraph](../_static/images/flamegraph.svg)
+
+To create a flamegraph, you need to install the `flamegraph` tool:
+
+```shell
+cargo install cargo-flamegraph
```
Comment (Rachelint) [inline docs/source/library-user-guide/profiling.md:40]:
```suggestion
cargo install flamegraph
```

File: docs/source/library-user-guide/profiling.md [hunk 2]
```diff
@@ -25,34 +25,44 @@ The section contains examples how to perform CPU profiling for Apache DataFusion
...
+Example: Flamegraph for a specific test:
 
 ```bash
-git clone https://github.com/brendangregg/FlameGraph.git /tmp/fg
-cd datafusion-cli
-CARGO_PROFILE_RELEASE_DEBUG=true cargo build --release
-echo "select * from table;" >> test.sql
-sudo dtrace -c './target/debug/datafusion-cli -f test.sql' -o out.stacks -n 'profile-997 /execname == "datafusion-cli"/ { @[ustack(100)] = count(); }'
-/tmp/fg/FlameGraph/stackcollapse.pl out.stacks | /tmp/fg/FlameGraph/flamegraph.pl > flamegraph.svg
+CARGO_PROFILE_RELEASE_DEBUG=true cargo flamegraph --root --unit-test datafusion  -- dataframe::tests::test_array_agg
```
Comment (Rachelint) [inline docs/source/library-user-guide/profiling.md:58]: Should we disable lto when profiling?
Seems it makes a small difference to flamegraph, and will lead to very very long building time...

Comment (alamb) [reply docs/source/library-user-guide/profiling.md:58]: I personally don't change the build settings when profiling -- it seems to work just fine for me

**Review by Rachelint — APPROVED** [review]
🚀  LGTM.
I try it following the instructions, and it works well.

**Comment by Rachelint** [comment]
Thanks @alamb

**Review by comphead — APPROVED** [review]
great thanks @alamb

**Comment by alamb** [comment]
Thank you @Rachelint and @comphead

**Pull request merged**